program-states/src/state/deposit_entry.rs

```diff
@@ -62,17 +63,27 @@ impl DepositEntry {
     /// Returns the weighted stake for the given deposit at the specified timestamp.
     #[inline(always)]
     pub fn weighted_stake(&self, curr_ts: u64) -> u64 {
-        if !self.is_used {
+        if !self.can_be_staked() {
```

Comment:
this sounds more like 
```suggestion
        if !self.is_staked() {
```